public/app/features/logs/components/panel/links.ts

```diff
@@ -0,0 +1,61 @@
+import { LinkModel } from '@grafana/data';
+
+import { FieldDef } from '../logParser';
+
+export function getTraceFromLinks(fields: FieldDef[]) {
+  for (const field of fields) {
+    if (!field.links) {
+      continue;
+    }
+    for (const link of field.links) {
+      const trace = getTraceFromLink(link);
+      if (trace) {
+        return trace;
+      }
+    }
+  }
+  return null;
+}
+
+function getTraceFromLink(link: LinkModel) {
+  const queryData = getDataSourceAndQueryFromLink(link);
+  if (!queryData || queryData.queryType !== 'traceql') {
+    return null;
+  }
+  return queryData;
+}
+
+export type EmbeddedInternalLink = {
+  dsUID: string;
+  query: string;
+  queryType: string;
+};
+
+function getDataSourceAndQueryFromLink(link: LinkModel): EmbeddedInternalLink | null {
+  if (!link.href) {
+    return null;
+  }
+  const paramsStrings = link.href.split('?')[1];
```

Comment:
Custom/external links might not have search params? Maybe we want to check for length first